soft4pes/control/lin/state_space_curr_ctr.py

```diff
@@ -0,0 +1,240 @@
+""" State-space current controller with anti-windup scheme 
+    for grid-connected converter with RL load """
+
+from types import SimpleNamespace
+import numpy as np
+from soft4pes.utils.conversions import alpha_beta_2_dq, dq_2_abc, dq_2_alpha_beta
+
+
+class RLGridStateSpaceCurrCtr:
+    """
+    State-space current controller with anti-windup scheme for grid-connected converter with RL load.
+    
+    Attributes
+    ----------
+    Rf : float
+        Resistance [p.u.].
+    Xf : float
+        Reactance [p.u.].
+    Ts : float
+        Sampling interval [s].           
+    Ts_pu : float
+        Sampling interval [p.u.].
+    ctr_pars : SimpleNamespace
+         A SimpleNamespace object containing controller parameters delta, K_i, k_ii and K_ti 
+    uc_ii_dq : 1 x 2 ndarray of floats
+        Converter voltage reference after current controller integrator in dq frame [p.u.].
+    uc_km1_dq : 1 x 2 ndarray of floats
+        Previous converter voltage reference in dq frame [p.u.].                                    
+    i_ref_seq_dq : Sequence
+        Current reference sequence instance in dq-frame [p.u.].   
+    sim_data : dict
+        Controller data.
+    """
+
+    def __init__(self, sys, base, Ts, i_ref_seq_dq):
+        """
+        Initialize a RLGridStateSpaceCurrCtr.
+
+        Parameters
+        ----------
+        sys : system object
+           System model.
+        base : base-value object
+           Base values.
+        Ts : float
+           Sampling interval [s].
+        i_ref_seq_dq : Sequence
+           Current reference sequence instance in dq-frame [p.u.].
+        """
+
+        self.Xf = sys.Xg  # Consider the filter inductane equals to the grid inductance
+        self.Rf = sys.Rg  # Consider the filter resitance equals to the grid resitance
+        self.Ts = Ts
+        self.Ts_pu = self.Ts * base.w
+        self.ctr_pars = self.get_state_space_ctr_pars()
+        self.u_ii_dq = np.zeros(2)
+        #In this controller uc_km1 is not considered due to not applying delay of PWM in the state-space model
```

Comment:
I do not understand what this comment is doing here. Also, it is too long. Check also linewidths on 94, 100, 103